pkg/apis/cloudcore/v1alpha1/types.go

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	metaconfig "github.com/kubeedge/kubeedge/pkg/apis/meta/v1alpha1"
+)
+
+type CloudCoreConfig struct {
+	metav1.TypeMeta
+	// KubeAPIConfig set the kubernetes cluster info which cloudcore will connected
+	// +Required
+	KubeAPIConfig KubeAPIConfig `json:"kubeAPIConfig,omitempty"`
+	// Modules set cloudcore modules config
+	// +Required
+	Modules Modules `json:"modules,omitempty"`
+}
+
+type KubeAPIConfig struct {
```

Comment:
Please add comment description for KubeAPIConfig

Reply:
done